pgmq-rs/tests/pg_ext_integration_test.rs

```diff
@@ -1,15 +1,35 @@
-use pgmq_core::{types::TABLE_PREFIX, util::connect};
+use pgmq_core::types::TABLE_PREFIX;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 use sqlx::{Pool, Postgres, Row};
 use std::env;
 
+// always test extension sdk in its own database
+// to avoid conflict with client only sdk
+fn replace_db_string(s: &str, replacement: &str) -> String {
+    match s.rfind('/') {
+        Some(pos) => {
+            let prefix = &s[0..pos];
+            format!("{}{}", prefix, replacement)
+        }
+        None => s.to_string(),
+    }
+}
+
 async fn init_queue_ext(qname: &str) -> pgmq::PGMQueueExt {
     let db_url = env::var("DATABASE_URL")
-        .unwrap_or_else(|_| "postgres://postgres:postgres@localhost:5432/pgmq".to_owned());
-    let queue = pgmq::PGMQueueExt::new(db_url, 2)
+        .unwrap_or_else(|_| "postgres://postgres:postgres@localhost:5432/postgres".to_owned());
+    let queue = pgmq::PGMQueueExt::new(db_url.clone(), 2)
         .await
         .expect("failed to connect to postgres");
+    // ignore error if d already exists
+    let _ = sqlx::query("CREATE DATABASE pgmq_ext_test;")
+        .execute(&queue.connection)
+        .await;
+    let test_db_str = replace_db_string(&db_url, "/pgmq_ext_test");
```

Comment:
running the extension API in its own database to avoid collisions between client-only and the extension test